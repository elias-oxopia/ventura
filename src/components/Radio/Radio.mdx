---
name: Radio
menu: Components
route: /radio
---

import { Playground, Props } from 'docz'
import { Radio } from '../../index.ts'

# Radio

Buttons make common actions more obvious and help users more easily perform them. Buttons use labels and sometimes icons to communicate the action that will occur when the user touches them.

<Playground>
  <Radio.Group name="class" defaultValue="0">
    <Radio.Item value="0" label="Java Fundementals" description="Get introduced to the basics of Java. Objects, primitive types and basic logic operators."/>
    <Radio.Item value="1" label="Advanced Web Development" description="Explore the different frameworks in the web."/>
    <Radio.Item value="2" label="User Experience Design" description="Learn to work with customer centricity in mind."/>
  </Radio.Group>
</Playground>

## API

<Props of={RadioGroup} />
