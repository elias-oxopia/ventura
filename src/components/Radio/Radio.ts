import RadioGroup from './RadioGroup/RadioGroup';
import RadioItem from './RadioItem/RadioItem';

export default { Group: RadioGroup, Item: RadioItem };
