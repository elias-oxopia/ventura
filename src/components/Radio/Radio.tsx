import React from 'react';

export interface Props {
  data?: string;
  size?: string | number;
}

const Radio: React.FC<Props> = ({ size = 24, color = 'currentColor' }: Props) => {
  return (
    <>
      {size}
      {color}
    </>
  );
};

export default Radio;
