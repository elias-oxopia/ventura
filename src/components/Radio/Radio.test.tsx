import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import Radio from './Radio';

describe('Radio', () => {
  test('default snapshot', () => {
    const component = (
      <Radio.Group name="default" value="test1">
        <Radio.Item value="test1" label="Test option 1" />
        <Radio.Item value="test2" label="Test option 2" />
        <Radio.Item value="test3" label="Test option 3" />
      </Radio.Group>
    );
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  test('description snapshot', () => {
    const component = (
      <Radio.Group name="default" value="test1">
        <Radio.Item value="test1" label="Test option 1" description="Lalalalalalala" />
        <Radio.Item value="test2" label="Test option 2" description="Lololololololo" />
        <Radio.Item value="test3" label="Test option 3" description="Lelelelelelele" />
      </Radio.Group>
    );
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  test('onChange should trigger correctly', () => {
    const onChangeFn = jest.fn();

    const component = (
      <Radio.Group name="default" value="test1" onChange={onChangeFn}>
        <Radio.Item value="test1" label="Test option 1" />
        <Radio.Item value="test2" label="Test option 2" />
        <Radio.Item value="test3" label="Test option 3" />
      </Radio.Group>
    );
    const { getByTestId } = render(component);

    const radioButton1 = getByTestId('radio-default-test1');
    const radioButton2 = getByTestId('radio-default-test2');
    const radioButton3 = getByTestId('radio-default-test3');
    fireEvent.click(radioButton1);
    fireEvent.click(radioButton2);
    fireEvent.click(radioButton3);
    fireEvent.click(radioButton3);
    fireEvent.click(radioButton2);
    fireEvent.click(radioButton1);

    expect(onChangeFn).toHaveBeenCalledTimes(4);
  });

  test('disabled should block onChange', () => {
    const onChangeFn = jest.fn();

    const component = (
      <Radio.Group name="default" value="test1" onChange={onChangeFn}>
        <Radio.Item isDisabled value="test1" label="Test option 1" />
        <Radio.Item isDisabled value="test2" label="Test option 2" />
        <Radio.Item isDisabled value="test3" label="Test option 3" />
      </Radio.Group>
    );
    const { getByTestId } = render(component);

    const radioButton1 = getByTestId('radio-default-test1');
    const radioButton2 = getByTestId('radio-default-test2');
    const radioButton3 = getByTestId('radio-default-test3');
    fireEvent.click(radioButton1);
    fireEvent.click(radioButton2);
    fireEvent.click(radioButton3);

    expect(onChangeFn).toHaveBeenCalledTimes(0);
  });

  test('className prop', () => {
    const className = 'center';
    const component = (
      <Radio.Group name="default" value="test1">
        <Radio.Item className={className} value="test1" label="Test option 1" />
        <Radio.Item value="test2" label="Test option 2" />
        <Radio.Item value="test3" label="Test option 3" />
      </Radio.Group>
    );
    const { getByTestId } = render(component);
    const containerElement = getByTestId('default-test1-container');

    const renderedClassNames = containerElement.className.split(' ');
    expect(renderedClassNames).toContain(className);
    // className in prop should be the last in the row
    expect(renderedClassNames.indexOf(className)).toBe(renderedClassNames.length - 1);
  });
});
