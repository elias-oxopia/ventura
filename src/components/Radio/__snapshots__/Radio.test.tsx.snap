// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio default snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <input
      checked=""
      class="radioInput"
      data-testid="radio-default-test1"
      id="default-test1"
      name="default"
      type="radio"
      value="test1"
    />
    <label
      class="radioButtonContainer radioButtonContainerChecked"
      data-testid="default-test1-container"
      for="default-test1"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 1
        </span>
      </div>
    </label>
    <input
      class="radioInput"
      data-testid="radio-default-test2"
      id="default-test2"
      name="default"
      type="radio"
      value="test2"
    />
    <label
      class="radioButtonContainer"
      data-testid="default-test2-container"
      for="default-test2"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 2
        </span>
      </div>
    </label>
    <input
      class="radioInput"
      data-testid="radio-default-test3"
      id="default-test3"
      name="default"
      type="radio"
      value="test3"
    />
    <label
      class="radioButtonContainer"
      data-testid="default-test3-container"
      for="default-test3"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 3
        </span>
      </div>
    </label>
  </div>
</DocumentFragment>
`;

exports[`Radio description snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <input
      checked=""
      class="radioInput"
      data-testid="radio-default-test1"
      id="default-test1"
      name="default"
      type="radio"
      value="test1"
    />
    <label
      class="radioButtonContainer radioButtonContainerChecked"
      data-testid="default-test1-container"
      for="default-test1"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 1
        </span>
      </div>
      <p
        class="description"
      >
        Lalalalalalala
      </p>
    </label>
    <input
      class="radioInput"
      data-testid="radio-default-test2"
      id="default-test2"
      name="default"
      type="radio"
      value="test2"
    />
    <label
      class="radioButtonContainer"
      data-testid="default-test2-container"
      for="default-test2"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 2
        </span>
      </div>
      <p
        class="description"
      >
        Lololololololo
      </p>
    </label>
    <input
      class="radioInput"
      data-testid="radio-default-test3"
      id="default-test3"
      name="default"
      type="radio"
      value="test3"
    />
    <label
      class="radioButtonContainer"
      data-testid="default-test3-container"
      for="default-test3"
    >
      <div
        class="header"
      >
        <div
          class="checkBox"
        >
          <div
            class="checkBoxFilling"
          />
        </div>
        <span
          class="title"
        >
          Test option 3
        </span>
      </div>
      <p
        class="description"
      >
        Lelelelelelele
      </p>
    </label>
  </div>
</DocumentFragment>
`;
